dcos_test_utils/dcos_api.py

```diff
@@ -259,15 +259,22 @@ def _wait_for_dcos_history_data(self):
         ro = self.get('/dcos-history-service/history/last')
         # resp_code >= 500 -> backend is still down probably
         if ro.status_code <= 500:
-            log.info("DC/OS History is probably getting data")
             json = ro.json()
-            # if an agent was removed, it may linger in the history data
-            assert len(json["slaves"]) >= len(self.all_slaves)
-            return True
-        else:
-            msg = "Waiting for DC/OS History, resp code is: {}"
-            log.info(msg.format(ro.status_code))
-            return False
+            # We have observed cases of the returned JSON being empty.
```

Comment:
What does "JSON being empty" mean? Did you observe the following JSON document, directly?
```
{}
```

Or, alternatively, is did you only see that the `slaves` key was not present?

Reply:
> What does "JSON being empty" mean? Did you observe the following JSON document, directly?

As per https://jira.mesosphere.com/browse/DCOS-18971?focusedCommentId=121689&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-121689 I saw:

```
<Response [200]>
(Pdb++) ro.json()
{}
```

I will change this message to be clearer.